### PR TITLE
docs: add README.md

### PR DESCRIPTION
**Pull request by coolaj86**
Because... well, y'know... every project deserves a README.

Preview at <https://github.com/coolaj86/rust-for-rustaceans.com/tree/patch-2>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Source for [rust-for-rustaceans.com](https://rust-for-rustaceans.com)
+
+Home of Reviews and Errata for ["Rust for Rustaceans" by Jon Gjengset](https://nostarch.com/rust-rustaceans).
+
+<kbd><img src="https://github.com/jonhoo/rust-for-rustaceans.com/blob/main/RustforRustaceans_cover.png?raw=true" title="Rust for Rustaceans" alt="book cover art with excited Ferris (Rust mascot)" /></kbd>
```
Comment (jonhoo) [inline README.md]:
```suggestion
![book cover art with excited Ferris (Rust mascot)](RustforRustaceans_cover.png?raw=true "Rust for Rustaceans")
```
I think this should work

File: README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+# Submit Errata
+
+Found a bug?
+
+Just take a look at [accepted Errata pull requests](https://github.com/jonhoo/rust-for-rustaceans.com/pulls?q=is%3Apr+is%3Aclosed+errata) and submit in similar form.
+
+## Example
+
+Branch / Commit / PR name: "errata print01-ch{nn}-lst{n} syntax error in example"
```
Comment (jonhoo) [inline README.md]:
```suggestion
Branch / Commit / PR name: "errata print01-ch{nn}-lst{nn} syntax error in example"
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+`````mkdn
```
Comment (jonhoo) [inline README.md]:
```suggestion
```markdown
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+---
+chapter: 7
+page: 103
+kind: code
+reporter: AJ ONeal
```
Comment (jonhoo) [inline README.md]:
```suggestion
reporter: Your name/any identifier you may prefer
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+## Example
+
+Branch / Commit / PR name: "errata print01-ch{nn}-lst{n} syntax error in example"
+
+`````mkdn
+---
+chapter: 7
+page: 103
+kind: code
```
Comment (jonhoo) [inline README.md:21]:
```suggestion
kind: code/typo
```

**Review by jonhoo — CHANGES_REQUESTED** [review]
This is a great idea, thanks for submitting it! I left some notes.

File: README.md (hunk 1 above)
Comment (coolaj86) [reply README.md]: It looks much better with the border. But, as you wish.

File: README.md (hunk 5 above)
Comment (coolaj86) [reply README.md:21]: Is this to suggest that "code/typo" is a category or that it may be "code" or "typo"?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+Found a bug?
+
+Just take a look at [accepted Errata pull requests](https://github.com/jonhoo/rust-for-rustaceans.com/pulls?q=is%3Apr+is%3Aclosed+errata) and submit in similar form.
+
+## Example
+
+Branch / Commit / PR name: "errata print01-ch{nn}-lst{nn} syntax error in example"
+
+````markdown
```
Comment (coolaj86) [inline README.md:17]: Note: extra backticks `` ```` `` because the body also contains backticks.

File: README.md (hunk 5 above)
Comment (coolaj86) [reply README.md:21]: I found the answer:

```bash
rg 'kind:'
```

Updated the README accordingly.

**Comment by coolaj86** [comment]
For reference:

No border: the white bleeds into the README, especially on normal monitors that don't have such fine distinction between shades of light grey as a Mac:

Screen Shot 2022-01-14 at 10 30 11 AM

With border: clearly contained image unto itself.

<kbd>Screen Shot 2022-01-14 at 10 30 11 AM<kbd>

**Comment by coolaj86** [comment]
Updated changes at: https://github.com/coolaj86/rust-for-rustaceans.com/tree/patch-2

File: README.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+```
+````
+
+Note:
+
+- `reporter` can be any identifier you wish, such as
+  - a name or nickname (ex: Jon Gjengset)
+  - a username or handle (ex: @jonhoo)
+  - perhaps an email, url, etc
+- `kind` may be one of
```
Comment (jonhoo) [inline README.md:53]: Currently `kind` isn't actually interpreted in any way, so any value goes. If I did ever ascribe semantics to them, I'd probably go through and fix up all the existing errata anyway. But having a listing doesn't hurt :+1:

File: README.md (hunk 6 above)
Comment (jonhoo) [reply README.md:17]: Ah, TIL that markdown lets you increase the number of fences in this way!

**Review by jonhoo — APPROVED** [review]
Excellent, thank you!

I'm not too concerned with the lack of a border in the README I'd rather use the right semantic HTML and clean Markdown. Styling markdown by (ab)using HTML elements doesn't seem worthwhile :)

**Pull request merged**